File: backends/php-laravel/app/Domain/Client/Repositories/JwtHistoryRepository.php
<?php

/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */

declare(strict_types=1);

namespace App\Domain\Client\Repositories;

use App\Domain\Client\Condition\JwtHistoryCondition;
use App\Domain\Client\Entities\JwtHistory;
use Illuminate\Support\Collection;

/**
 * JWT履歴Repositoryのインターフェースです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 * @package App\Domain\Client\Repositories
 */
interface JwtHistoryRepository
{
    /**
     * クライアントIDに紐づくJWT履歴を返します。
     *
     * @param JwtHistoryCondition $condition 検索条件
     * @return Collection エンティティコレクション
     */
    public function findByClientId(JwtHistoryCondition $condition): Collection;

    /**
     * JWT履歴を保存します。
     *
     * @param JwtHistory $entity エンティティ
     * @return JwtHistory 保存後のエンティティ
     */
    public function persist(JwtHistory $entity): JwtHistory;
}

File: backends/java-spring/src/main/java/com/authorization/domain/client/repositories/JwtHistoryRepository.java
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
package com.authorization.domain.client.repositories;

import com.authorization.domain.client.condition.JwtHistoryCondition;
import com.authorization.domain.client.entities.JwtHistory;
import java.util.List;

/**
 * JWT履歴Repositoryのインターフェースです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
public interface JwtHistoryRepository {

    /**
     * クライアントIDに紐づくJWT履歴の総件数を返します。
     *
     * @param condition 検索条件
     * @return 総件数
     */
    int countByClientId(JwtHistoryCondition condition);

    /**
     * クライアントIDに紐づくJWT履歴を返します。
     *
     * @param condition 検索条件
     * @return エンティティ一覧
     */
    List<JwtHistory> findByClientId(JwtHistoryCondition condition);

    /**
     * JWT履歴を保存します。
     *
     * @param entity エンティティ
     * @return 保存後のエンティティ
     */
    JwtHistory persist(JwtHistory entity);
}
